pkg/storage/bucket/s3/config.go

```diff
@@ -107,6 +108,7 @@ func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.ListObjectsVersion, prefix+"s3.list-objects-version", "", "Use a specific version of the S3 list object API. Supported values are v1 or v2. Default is unset.")
 	f.StringVar(&cfg.StorageClass, prefix+"s3.storage-class", "", "The S3 storage class to use, not set by default. Details can be found at https://aws.amazon.com/s3/storage-classes/. Supported values are: "+strings.Join(supportedStorageClasses, ", "))
 	f.BoolVar(&cfg.NativeAWSAuthEnabled, prefix+"s3.native-aws-auth-enabled", false, "If enabled, it will use the default authentication methods of the AWS SDK for go based on known environment variables and known AWS config files.")
+	f.Uint64Var(&cfg.PartSize, prefix+"s3.part-size", s3.DefaultConfig.PartSize, "The minimum file size in bytes used for multipart uploads. If 0, a built-in default value is used.")
```

Comment:
we're using this 64MiB default from `s3.DefaultConfig.PartSize` but I believe that we weren't using this in Mimir and were using the value of `0`. 

https://github.com/grafana/mimir/blob/14622a8a3e536cc03d0a46833b02bcdd5a38f1e6/pkg/storage/bucket/s3/bucket_client.go#L52-L76

what I could find was in the minio client which was using a default of 16MiB in some cases and sometimes this is configured based on the object size

https://github.com/grafana/mimir/blob/3d62b188ff7a99be0345b8e7e1f721eb7b071a48/vendor/github.com/minio/minio-go/v7/api-put-object.go#L296-L299
https://github.com/grafana/mimir/blob/3d62b188ff7a99be0345b8e7e1f721eb7b071a48/vendor/github.com/minio/minio-go/v7/constants.go#L28
https://github.com/grafana/mimir/blob/d99cb3c081022e0752ee3db8c2cc39af46d09ba8/vendor/github.com/minio/minio-go/v7/api-put-object-common.go#L116-L120

Having said that I think it's safer if we have `0` as the default for Mimir's flag and update the help text to say that if we're at 0, then the value is optimally computed for each object

Reply:
Thank you for your feedback!

> I believe that we weren't using this in Mimir and were using the value of 0.

Correct me if I am wrong, but you weren't passing `PartSize`, so `thanos-io/objstrore` were setting the default value here:

https://github.com/grafana/mimir/blob/dbc941a31aa05a3420e21e6f4ccd7a67321cd7fb/vendor/github.com/thanos-io/objstore/providers/s3/s3.go#L171-L179

If we set `0`, it will still set the default, right?

Reply:
Ohh, no. You are using `NewBucketWithConfig` so `parseConfig` is not executed, and it is set to `0`. Sure, I will change the default value to `0`. Thank you!